pkg/dio/writer.go

```diff
@@ -49,34 +50,41 @@ func (t *multiWriter) Write(p []byte) (int, error) {
 	}
 
 	t.mutex.Lock()
+
+	t.waitGroup.Add(1)
```

Comment:
There is a race here where if Close() gets the lock first, it will close and clear out the writers. When Write() gets through, it is left with no writers, and just returns without writing anything (and no error). This is borderline acceptable; the wait group does not prevent this case. It appears to be safe to use though, although I think just grabbing the lock for the duration of the writes would have the same affect.